smos-cursor/src/Smos/Cursor/Tag.hs

```diff
@@ -80,3 +81,8 @@ tagCursorSelectPrevChar = tagCursorTextCursorL textCursorSelectPrev
 
 tagCursorSelectNextChar :: TagCursor -> Maybe TagCursor
 tagCursorSelectNextChar = tagCursorTextCursorL textCursorSelectNext
+
+tagCursorSplit :: TagCursor -> (TagCursor, TagCursor)
+tagCursorSplit = tagCursorTextCursorL $ \tc ->
+  let (a, b) = textCursorSplit tc
+   in (TagCursor a, b)
```

Comment:
I would _really_ prefer if we don't use the `Functor` instance of tuples, but this is fine, I guess.
We may want to just add a comment about that.

Reply:
Is there any better way to do it instead of using tuples?

Reply:
You can do explicit wrapping and unwrapping. They're just newtypes anyway.

Reply:
Do you mean something like this?
```haskell
tagCursorSplit (TagCursor tc) =
  let (a, b) = textCursorSplit tc
   in (TagCursor a, TagCursor b)
```

Reply:
Yes that's right.